src/TSTransformer/nodes/expressions/transformTemplateExpression.ts

```diff
@@ -1,29 +1,28 @@
 import luau from "@roblox-ts/luau-ast";
 import { errors } from "Shared/diagnostics";
-import { assert } from "Shared/util/assert";
 import { TransformState } from "TSTransformer";
 import { DiagnosticService } from "TSTransformer/classes/DiagnosticService";
 import { createStringFromLiteral } from "TSTransformer/util/createStringFromLiteral";
 import { ensureTransformOrder } from "TSTransformer/util/ensureTransformOrder";
-import { binaryExpressionChain } from "TSTransformer/util/expressionChain";
-import {
-	isDefinitelyType,
-	isLuaTupleType,
-	isPossiblyType,
-	isStringType,
-	isUndefinedType,
-} from "TSTransformer/util/types";
+import { isDefinitelyType, isLuaTupleType } from "TSTransformer/util/types";
 import ts from "typescript";
 
-export function transformTemplateExpression(state: TransformState, node: ts.TemplateExpression) {
-	// if there are zero templateSpans, this must be a ts.NoSubstitutionTemplateLiteral
-	// and will be handled in transformStringLiteral
-	assert(node.templateSpans.length > 0);
+export function transformTemplateExpression(
+	state: TransformState,
+	node: ts.TemplateExpression | ts.NoSubstitutionTemplateLiteral,
+) {
+	// backtick string literals without interpolation expressions should be preserved
+	// as they still are valid in luau
+	if (ts.isNoSubstitutionTemplateLiteral(node)) {
+		return luau.create(luau.SyntaxKind.InterpolatedString, {
+			segments: luau.list.make(createStringFromLiteral(node)),
+		});
+	}
```

Comment:
Is it an improvement to merge this special case into this function, when it still early returns?